src/test/ui/moves/use_of_moved_value_clone_suggestions.stderr

```diff
@@ -1,13 +1,71 @@
 error[E0382]: use of moved value: `t`
-  --> $DIR/use_of_moved_value_clone_suggestions.rs:3:9
+  --> $DIR/use_of_moved_value_clone_suggestions.rs:7:9
    |
-LL | fn duplicate_rc<T>(t: std::rc::Rc<T>) -> (std::rc::Rc<T>, std::rc::Rc<T>) {
-   |                    - move occurs because `t` has type `Rc<T>`, which does not implement the `Copy` trait
-LL |     (t, t)
+LL | fn move_rc<T>(t: std::rc::Rc<T>) {
+   |               - move occurs because `t` has type `Rc<T>`, which does not implement the `Copy` trait
+LL |     [t, t];
    |      -  ^ value used here after move
    |      |
    |      value moved here
+   |
+help: consider cloning `t`
+   |
+LL |     [t.clone(), t];
+   |       ++++++++
+
+error[E0382]: use of moved value: `t`
+  --> $DIR/use_of_moved_value_clone_suggestions.rs:13:9
+   |
+LL | fn move_clone_already<T: Clone>(t: T) {
+   |                                 - move occurs because `t` has type `T`, which does not implement the `Copy` trait
+LL |     [t, t];
+   |      -  ^ value used here after move
+   |      |
+   |      value moved here
+   |
+help: consider cloning `t`
+   |
+LL |     [t.clone(), t];
+   |       ++++++++
+
+error[E0382]: use of moved value: `t`
+  --> $DIR/use_of_moved_value_clone_suggestions.rs:18:9
+   |
+LL | fn move_clone_only<T>(t: (T, String)) {
+   |                       - move occurs because `t` has type `(T, String)`, which does not implement the `Copy` trait
+LL |     [t, t];
+   |      -  ^ value used here after move
+   |      |
+   |      value moved here
+   |
+help: consider restricting type parameter `T`
+   |
+LL | fn move_clone_only<T: Clone>(t: (T, String)) {
+   |                     +++++++
+help: ...and cloning `t`
+   |
+LL |     [t.clone(), t];
+   |       ++++++++
```

Comment:
To be clear: these suggestions could be merged together, it just requires some work (we need the `suggest_constraining_type_params` to allow adding to the same suggestion/changing the text/etc).